SMLHelper/Patchers/LootDistributionPatcher.cs

```diff
@@ -0,0 +1,119 @@
+                        {
+                            bool foundPrefab = false;
+
+                            for (int j = 0; j < dstData.prefabs.Count; j++)
+                            {
+                                LootDistributionData.PrefabData prefabData = dstData.prefabs[j];
+
+                                if (prefabData.classId == classId)
+                                {
+                                    prefabData.count = customBiomeDist.count;
+                                    prefabData.probability = customBiomeDist.probability;
+
+                                    foundPrefab = true;
+                                }
+                            }
+
+                            if(!foundPrefab)
+                            {
+                                dstData.prefabs.Add(new LootDistributionData.PrefabData()
+                                {
+                                    classId = classId,
+                                    count = customBiomeDist.count,
+                                    probability = customBiomeDist.probability
+                                });
+                            }
+                        }
+                    }
+                }
+                else
+                {
+                    __instance.srcDistribution.Add(entry.Key, entry.Value);
+
+                    string classId = entry.Key;
+                    LootDistributionData.SrcData data = entry.Value;
+                    List<LootDistributionData.BiomeData> distribution = data.distribution;
+
+                    if (distribution != null)
+                    {
+                        for(int i = 0; i < distribution.Count; i++)
+                        {
+                            LootDistributionData.BiomeData biomeData = distribution[i];
+                            BiomeType biome = biomeData.biome;
+                            int count = biomeData.count;
+                            float probability = biomeData.probability;
+                            LootDistributionData.DstData dstData;
+
+                            if (!__instance.dstDistribution.TryGetValue(biome, out dstData))
+                            {
+                                dstData = new LootDistributionData.DstData();
+                                dstData.prefabs = new List<LootDistributionData.PrefabData>();
+                                __instance.dstDistribution.Add(biome, dstData);
+                            }
+
+                            LootDistributionData.PrefabData prefabData = new LootDistributionData.PrefabData();
+                            prefabData.classId = classId;
+                            prefabData.count = count;
+                            prefabData.probability = probability;
+                            dstData.prefabs.Add(prefabData);
+                        }
```

Comment:
To make this easier to read, refactor the code of the if and the else blocks into their own methods